CMakeLists.txt

```
@@ -1,25 +1,22 @@
 cmake_minimum_required(VERSION 3.4 FATAL_ERROR)
-PROJECT(sioclient)
+PROJECT(sioclient
+        VERSION 3.1.0
+        )
 
 option(BUILD_SHARED_LIBS "Build the shared library" OFF)
-option(BUILD_UNIT_TESTS  "Builds unit tests target" OFF)
+option(BUILD_UNIT_TESTS "Builds unit tests target" OFF)
 
-set(MAJOR 1)
-set(MINOR 6)
-set(PATCH 0)
-
-if(NOT CMAKE_BUILD_TYPE )
-MESSAGE(STATUS "not define build type, set to release" )
-set(CMAKE_BUILD_TYPE Release )
-elseif(NOT (${CMAKE_BUILD_TYPE} STREQUAL "Release" OR ${CMAKE_BUILD_TYPE} STREQUAL "Debug" ))
-MESSAGE(SEND_ERROR "CMAKE_BUILD_TYPE must be either Release or Debug")
-return()
-endif()
+if (NOT EXISTS ${CMAKE_SOURCE_DIR}/lib/asio/asio/README)
+    message("Updating submodules")
+    execute_process(
+            COMMAND git submodule update --init
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+endif ()
```

It is the developer responsibility to do this and not the CMake. We want to keep the CMakeLists.txt minimal if possible.

```suggestion
set(MAJOR 1)
set(MINOR 6)
set(PATCH 0)
```